boards/shared/Src/App/App_SharedErrorTable.c

```diff
@@ -0,0 +1,293 @@
+    {
+        error_table->errors[i] = App_SharedError_Create();
+    }
+
+    App_SharedError_SetBoard(error_table->errors[BMS_NON_CRIT], BMS);
+    App_SharedError_SetId(error_table->errors[BMS_NON_CRIT], BMS_NON_CRIT);
+    App_SharedError_SetIsCritical(error_table->errors[BMS_NON_CRIT], false);
+
+    App_SharedError_SetBoard(error_table->errors[DCM_NON_CRIT], DCM);
+    App_SharedError_SetId(error_table->errors[DCM_NON_CRIT], DCM_NON_CRIT);
+    App_SharedError_SetIsCritical(error_table->errors[DCM_NON_CRIT], false);
+
+    App_SharedError_SetBoard(error_table->errors[DIM_NON_CRIT], DIM);
+    App_SharedError_SetId(error_table->errors[DIM_NON_CRIT], DIM_NON_CRIT);
+    App_SharedError_SetIsCritical(error_table->errors[DIM_NON_CRIT], false);
+
+    App_SharedError_SetBoard(error_table->errors[FSM_NON_CRIT], FSM);
+    App_SharedError_SetId(error_table->errors[FSM_NON_CRIT], FSM_NON_CRIT);
+    App_SharedError_SetIsCritical(error_table->errors[FSM_NON_CRIT], false);
+
+    App_SharedError_SetBoard(error_table->errors[PDM_NON_CRIT], PDM);
+    App_SharedError_SetId(error_table->errors[PDM_NON_CRIT], PDM_NON_CRIT);
+    App_SharedError_SetIsCritical(error_table->errors[PDM_NON_CRIT], false);
+
+    App_SharedError_SetBoard(error_table->errors[BMS_CRIT], BMS);
+    App_SharedError_SetId(error_table->errors[BMS_CRIT], BMS_CRIT);
+    App_SharedError_SetIsCritical(error_table->errors[BMS_CRIT], true);
+
+    App_SharedError_SetBoard(error_table->errors[DCM_CRIT], DCM);
+    App_SharedError_SetId(error_table->errors[DCM_CRIT], DCM_CRIT);
+    App_SharedError_SetIsCritical(error_table->errors[DCM_CRIT], true);
+
+    App_SharedError_SetBoard(error_table->errors[DIM_CRIT], DIM);
+    App_SharedError_SetId(error_table->errors[DIM_CRIT], DIM_CRIT);
+    App_SharedError_SetIsCritical(error_table->errors[DIM_CRIT], true);
+
+    App_SharedError_SetBoard(error_table->errors[FSM_CRIT], FSM);
+    App_SharedError_SetId(error_table->errors[FSM_CRIT], FSM_CRIT);
+    App_SharedError_SetIsCritical(error_table->errors[FSM_CRIT], true);
+
+    App_SharedError_SetBoard(error_table->errors[PDM_CRIT], PDM);
+    App_SharedError_SetId(error_table->errors[PDM_CRIT], PDM_CRIT);
+    App_SharedError_SetIsCritical(error_table->errors[PDM_CRIT], true);
+
+    return error_table;
+}
+
+void App_SharedErrorTable_Destroy(struct ErrorTable *error_table)
+{
+    for (size_t i = 0; i < NUM_ERROR_IDS; i++)
+    {
+        App_SharedError_Destroy(error_table->errors[i]);
+    }
+
+    free(error_table);
+}
+
+ExitCode App_SharedErrorTable_SetError(
+    struct ErrorTable *error_table,
+    enum ErrorId       error_id,
+    bool               is_set)
+{
+    if ((int)error_id >= NUM_ERROR_IDS)
+    {
+        return EXIT_CODE_OUT_OF_RANGE;
+    }
+    struct Error *error = error_table->errors[error_id];
+    App_SharedError_SetIsSet(error, is_set);
+
+    return EXIT_CODE_OK;
+}
+
+bool App_SharedErrorTable_HasError(const struct ErrorTable *error_table)
```

Comment:
Just to be super clear, perhaps we could call this `HasAnyErrorSet` ?

Reply:
Updated `HasCriticalError` and `HasNonCriticalError` too.